python-threatexchange/threatexchange/cli/match.py

```diff
@@ -155,7 +155,9 @@ def execute(self, api: ThreatExchangeAPI, dataset: Dataset) -> None:
         )
 
         file_matchers = [s for s in all_signal_types if isinstance(s, FileMatcher)]
-        str_matchers = [s for s in all_signal_types if isinstance(s, StrMatcher)]
+        str_matchers: t.List[t.Any] = [
```

Comment:
sounds like all str_matchers will be StrMatcher instances. To note t.List is an [invariant type](https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance). If you are having trouble with typing this, use `t.Sequence[StrMatcher]`.

Reply:
I tried but both using `t.Sequence` and `t.Sequence[t.Union[...]]` fall apart on line 173 of this file and I don't want to change too much of the logic in one PR.